### PR TITLE
feat(tensorflow frontend): add max_pool3d and its test function

### LINKED ISSUE #26480: MaxPool3D

### PR DESCRIPTION
**Pull request by phattrann**
Close #26480 

# PR Description
Implement **MaxPool3D** in RawOps operation for Tensorflow Frontend.
The function might have identical behavior to [ivy/ivy/functional/frontends/tensorflow/nn.py](https://github.com/unifyai/ivy/blob/main/ivy/functional/frontends/tensorflow/nn.py) so I implement the function in the `nn.py` and add the alias in the [ivy/ivy/functional/frontends/tensorflow/raw_ops.py](https://github.com/unifyai/ivy/blob/main/ivy/functional/frontends/tensorflow/raw_ops.py).

After running several tests with different datatypes. I found out that only the `float32` dtype worked well on all 5 backend tests.
`bfloat16` and `float16` not working on jax and `float16` not working on numpy.

## Checklist

- [x] Did you add a function?
- [x] Did you add the tests?
- [x] Did you run your tests and are your tests passing?
- [x] Did pre-commit not fail on any check?
- [x] Did you follow the steps we provided?

### CODE REVIEW

**Comment by phattrann** [comment]
Hi @hirwa-nshuti, Do I have to write the test in the `test_raw_ops.py` for the **alias** as well? Because I only wrote the test for the `test_nn.py`. If it is, then I'm missing 1 test.

**Comment by hirwa-nshuti** [comment]
> Hi @hirwa-nshuti, Do I have to write the test in the `test_raw_ops.py` for the **alias** as well? Because I only wrote the test for the `test_nn.py`. If it is, then I'm missing 1 test.

Yeah you have to write the tests for the `raw_ops` version as well

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Hey @phattrann 

Would you add the unit tests for the `raw_ops` version.
The rest looks great to me

**Comment by phattrann** [comment]
Hi @hirwa-nshuti, I wrote the unit test for `MaxPool3D` in the `raw_ops` but I've been stuck for a day, and still have no clue. What I know so far is that I got 2 errors from 2 generated inputs by the hypothesis_test:

- Input: `ksize: (1,1,1)`
Error: `tensorflow.python.framework.errors_impl.InvalidArgumentError: Length for attr 'ksize' of 3 must be at least minimum 5`
- Input:`strides: (1,)`
Error: `tensorflow.python.framework.errors_impl.InvalidArgumentError: Length for attr 'strides' of 1 must be at least minimum 5`

Meaning that `ksize` and `strides` should be a tuple (or list) with length of 5 such as `(1, 1, 1, 1, 1)`.

I have debugged on the max_pool3d in `test_nn.py` version and both inputs with the same length worked really well. But I don't know what is wrong with the alias `test_raw_ops.py` version. Please review my code again. Thank you!

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_nn.py [hunk 1]
```diff
@@ -1497,6 +1497,40 @@ def test_tensorflow_max_pool2d(
     )
 
 
+# max_pool3d
+@handle_frontend_test(
+    fn_tree="tensorflow.nn.max_pool3d",
+    data_format=df(data_format=st.sampled_from(["NDHWC"])),
```
Comment (hirwa-nshuti) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_nn.py:1503]: You can simply here use `data_format=st.sampled_from(["NDHWC", "NCDHW"])`

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Hey @phattrann 
Check the comment below and update your branch with main branch for CI checks to run
Thanks!

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_nn.py (hunk 1 above)
Comment (phattrann) [reply ivy_tests/test_ivy/test_frontends/test_tensorflow/test_nn.py]: Thanks @hirwa-nshuti, I have applied that for both test functions.

**Comment by phattrann** [comment]
Hi @hirwa-nshuti, I have updated the branch and got CI checks done, please have a look at my PR once again. Thank you!

**Review by hirwa-nshuti — APPROVED** [review]
Looks good to me😊
Thanks for contributing

**Comment by phattrann** [comment]
Thank you for your reviewing @hirwa-nshuti

**Pull request merged**